eclipse-collections-code-generator/src/main/resources/impl/list/mutable/synchronizedPrimitiveList.stg

```diff
@@ -409,6 +413,15 @@ public class Synchronized<name>List
             return this.getMutable<name>List().collectWithIndex(function, target);
         }
     }
+<if(primitive.specializedStream)>
+    @Override
+    public Spliterator.Of<name> primitiveSpliterator(){
+        synchronized (this.getLock())
+        {
+            return this.getMutable<name>List().primitiveSpliterator();
+        }
+    }
```

Comment:
Here are the relevant pieces of Javadoc. Let me know if it needs clarification.

```java
/**
 * A synchronized view of a {@link Mutable<name>List}. It is imperative that the user manually synchronize on the collection when iterating over it using the
 * {@link <name>Iterator}, as per {@link Collections#synchronizedCollection(Collection)}.
 * \<p>
 * This file was automatically generated from template file synchronizedPrimitiveList.stg.
 *
 * @see Mutable<name>List#asSynchronized()
 * @see MutableList#asSynchronized()
 * @since 3.1.
 */
```

```java
    /**
     * Returns a synchronized (thread-safe) collection backed by the specified
     * collection.  In order to guarantee serial access, it is critical that
     * <strong>all</strong> access to the backing collection is accomplished
     * through the returned collection.<p>
     *
     * It is imperative that the user manually synchronize on the returned
     * collection when traversing it via {@link Iterator}, {@link Spliterator}
     * or {@link Stream}:
     * <pre>
     *  Collection c = Collections.synchronizedCollection(myCollection);
     *     ...
     *  synchronized (c) {
     *      Iterator i = c.iterator(); // Must be in the synchronized block
     *      while (i.hasNext())
     *         foo(i.next());
     *  }
     * </pre>
     * Failure to follow this advice may result in non-deterministic behavior.
     *
     * <p>The returned collection does <i>not</i> pass the {@code hashCode}
     * and {@code equals} operations through to the backing collection, but
     * relies on {@code Object}'s equals and hashCode methods.  This is
     * necessary to preserve the contracts of these operations in the case
     * that the backing collection is a set or a list.<p>
     *
     * The returned collection will be serializable if the specified collection
     * is serializable.
     *
     * @param  <T> the class of the objects in the collection
     * @param  c the collection to be "wrapped" in a synchronized collection.
     * @return a synchronized view of the specified collection.
     */
    public static <T> Collection<T> synchronizedCollection(Collection<T> c) {
        return new SynchronizedCollection<>(c);
    }
```

Reply:
Ignore my comment about throwing unsupported. I looked around and we never do that. We just warn about thread-safe access in Javadoc.

Reply:
ok. i removed the synchronized block